packages/clerk-js/src/ui/components/OrganizationProfile/DomainList.tsx

```diff
@@ -109,53 +105,43 @@ export const DomainList = withGate(
       return null;
     }
 
-    // TODO: Split this to smaller components
     return (
-      <Col>
+      <ProfileSection.ItemList id='organizationDomains'>
         {domainList.length === 0 && !domains?.isLoading && fallback}
-        {domainList.map(d => {
-          if (!(d.verification && d.verification.status === 'verified') || !canManageDomain) {
-            return (
-              <BlockWithTrailingComponent
-                key={d.id}
-                sx={t => ({
-                  '&:hover': {
-                    backgroundColor: t.colors.$blackAlpha50,
-                  },
-                  padding: `${t.space.$none} ${t.space.$4}`,
-                  minHeight: t.sizes.$10,
-                })}
-                badge={<EnrollmentBadge organizationDomain={d} />}
-                trailingComponent={
-                  canManageDomain ? (
-                    <DomainListDotMenu
-                      redirectSubPath={redirectSubPath}
-                      domainId={d.id}
-                    />
-                  ) : undefined
-                }
-              >
-                {d.name}
-              </BlockWithTrailingComponent>
-            );
-          }
-
+        {domainList.map(domain => {
           return (
-            <ArrowBlockButton
-              key={d.id}
-              variant='ghost'
-              badge={!verificationStatus ? <EnrollmentBadge organizationDomain={d} /> : undefined}
-              sx={t => ({
-                padding: `${t.space.$3} ${t.space.$4}`,
-                minHeight: t.sizes.$10,
-              })}
-              onClick={() => navigate(buildDomainListRelativeURL(redirectSubPath, d.id))}
-            >
-              {d.name}
-            </ArrowBlockButton>
+            <Action.Root key={domain.id}>
+              <ProfileSection.Item id='organizationDomains'>
+                <Flex sx={t => ({ gap: t.space.$1 })}>
+                  <Text>{domain.name}</Text>
+                  <EnrollmentBadge organizationDomain={domain} />
+                </Flex>
+
+                {canManageDomain && <DomainListMenu domain={domain} />}
+              </ProfileSection.Item>
+
+              <Action.Open value='remove'>
+                <Action.Card>
+                  <RemoveDomainForm domainId={domain.id} />
+                </Action.Card>
+              </Action.Open>
+
+              <Action.Open value='verify'>
+                <Action.Card>
+                  <VerifyDomainForm domainId={domain.id} />
+                </Action.Card>
+              </Action.Open>
+
+              <Action.Open value='manage'>
+                <Action.Card>
+                  <VerifiedDomainForm domainId={domain.id} />
+                </Action.Card>
+              </Action.Open>
+            </Action.Root>
           );
         })}
-        {(domains?.hasNextPage || domains?.isFetching) && (
+
+        {(domains?.hasNextPage || domains?.isFetching) && domains.data.length === 0 && (
```

Comment:
Wouldn't this break the fetching of subsequent pages ?

Reply:
why